// @caseproof/eslint-plugin/lib/configs/base.js
module.exports = {
    env: {
        browser: true,
        es2020: true,
        node: true,
    },
    parser: '@typescript-eslint/parser',
    extends: [
        'eslint:recommended',
        'plugin:@typescript-eslint/recommended',
        'plugin:jest/recommended',
        'plugin:jsdoc/recommended-typescript',
    ],
    plugins: ['jest', 'jsdoc', '@typescript-eslint'],
    overrides: [
        {
            env: {
                'jest/globals': true,
            },
            files: [
                '**/__tests__/**/*.js',
            ],
        },
        {
            env: {
                node: true,
                browser: false,
            },
            files: [
                '.eslintrc.{js,cjs}',
                'jest.config.{js,cjs}',
                'babel.config.{js,cjs}',
                'postcss.config.{js,cjs}',
                'tailwind.config.{js,cjs}',
                'vite.config.{js,cjs}',
            ],
            parserOptions: {
                sourceType: 'script'
            }
        },
        {
            env: {
                node: true,
                browser: false,
            },
            files: [
                'vite.config.{js,cjs}',
            ],
            parserOptions: {
                sourceType: 'module'
            }
        }
    ],
    parserOptions: {
        ecmaVersion: 'latest',
        sourceType: 'module'
    },
    ignorePatterns: [
        'node_modules/',
        'vendor/',
        'public/',
        '**/*.d.ts',
    ],
    rules: {
        indent: [
            'error',
            4
        ],
        'linebreak-style': [
            'error',
            'unix'
        ],
        quotes: [
            'error',
            'single'
        ],
        semi: [
            'error',
            'always'
        ],
        'jsdoc/tag-lines': [
            'warn',
            'never',
            {
                'startLines': 1
            }
        ]
    }
};
